SUS-1276

mixth-sense opened the pull request.

Commit:
SUS-1276 → retire SpecialEditHub

Commit:
SUS-1276 → retire SpecialManageWikiaHome

Commit:
SUS-1276 → retire FormBuilderService

Commit:
SUS-1276 remove SpecialManageWikiaHome traces

WikiaTech commented:
SonarQube analysis reported 22 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [maintenance.php#L31](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L31): Class "AutomatedDeadWikisDeletionMaintenance" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [maintenance.php#L119](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L119): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L153](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L153): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [maintenance.php#L215](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L215): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L221](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L221): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L226](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L226): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L265](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L265): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [maintenance.php#L267](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L267): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [maintenance.php#L421](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L421): Remove this unused "$statusCode" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [maintenance.php#L438](https://github.com/Wikia/app/blob/eb349915bb559ba1cbe135813fc0f14ef73a5903/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L438): Add a "case default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SUS-1276 → retire FlashMessages

WikiaTech commented:
SonarQube analysis reported 22 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 1 minor
* ![INFO][INFO] 5 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [maintenance.php#L31](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L31): Class "AutomatedDeadWikisDeletionMaintenance" has 24 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [maintenance.php#L119](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L119): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L153](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L153): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [maintenance.php#L215](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L215): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L221](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L221): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L226](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L226): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [maintenance.php#L265](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L265): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [maintenance.php#L267](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L267): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [maintenance.php#L421](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L421): Remove this unused "$statusCode" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [maintenance.php#L438](https://github.com/Wikia/app/blob/a3c3d27f0e8005129804489f5f20c63a3aaa73fd/extensions/wikia/WikiFactory/DeadWikis/maintenance.php#L438): Add a "case default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Commit:
SUS-1276 → remove unused WikiaHopePageHelper::getNumberOfSlotsForType()

Commit:
SUS-1276 → remove unused WikiaHopePageHelper methods

Commit:
SUS-1276 → WikiaHomePageHelper remove unused constants

Commit:
SUS-1276 → reduce WikiaHomePageHelper

Move WikiaHomePageHelper's methods that are used only once in the code to
classes that use them.

Commit:
SUS-1276 → stripping WikiaHomePageHelper and CityVisualisation, cont'd.

WikiaTech commented on extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php:
```diff
@@ -14,9 +14,6 @@
 
 class WikiaHomePageHelper extends WikiaModel {
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Class "WikiaHomePageHelper" has 36 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)

WikiaTech commented:
SonarQube analysis reported 75 issues
* ![MAJOR][MAJOR] 46 major
* ![MINOR][MINOR] 18 minor
* ![INFO][INFO] 11 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L78](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L78): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L88](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L88): Replace this legacy wfMsg* function call with one of the new i18n functions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3ALegacyWfMsgUsage)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L133](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L133): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L298](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L298): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L511](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L511): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L520](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L520): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L7](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L7): Class "CityVisualization" has 51 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L97](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L97): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L173](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L173): Remove this unused "$verticalsCount" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L205](https://github.com/Wikia/app/blob/04d7f33eeb2f178d1ca71e28ff7f07230b8263bc/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L205): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

mszabo-wikia commented on includes/wikia/FlashMessages.class.php:
```diff
@@ -1,17 +0,0 @@
-		global $wgRequest;
-		$wgRequest->setSessionData( self::FLASH_MESSAGE_SESSION_KEY, $message );
-	}
-}
```
![WTF](https://i.imgur.com/sSJQgmW.gif)

mszabo-wikia reviewed (approved):
I was meaning to ask if the corresponding tests have also been removed... but it seems there weren't any to begin with. 😞

Commit:
SUS-1276 → stripping WikiaHomePageHelper and CityVisualisation, cont'd.

WikiaTech commented:
SonarQube analysis reported 74 issues
* ![MAJOR][MAJOR] 45 major
* ![MINOR][MINOR] 18 minor
* ![INFO][INFO] 11 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L78](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L78): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L129](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L129): Add curly braces around the nested statement(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L215](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L215): Reduce the number of conditional operators (4) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1067)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L422](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L422): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [WikiaHomePageHelper.class.php#L431](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/helpers/WikiaHomePageHelper.class.php#L431): Remove this commented out code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS125)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L7](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L7): Class "CityVisualization" has 51 methods, which is greater than 20 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L97](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L97): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L173](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L173): Remove this unused "$verticalsCount" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L205](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L205): Replace or remove this deprecated Wikia::log logging statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AWikiaLogUsage)
1. ![MAJOR][MAJOR] [CityVisualization.class.php#L430](https://github.com/Wikia/app/blob/02fe18ca8504e6cc292a73e702aed4034cd58e22/extensions/wikia/CityVisualization/models/CityVisualization.class.php#L430): Remove this unused private "getVerticalNameId" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1144)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
